### PR TITLE
Patch quotes

### LINKED ISSUE #2: CSV doesn't observe quoted strings.
When using a CSV format that has quoted values that contain a comma the tool splits the value into columns.

Given

```
1,2,3
a,b,c
a,"b,b",c
```

Expected:

| 1 | 2 | 3 |
| --- | --- | --- |
| a | b | c |
| a | b,b | c |

Actual

| 1 | 2 | 3 |  |
| --- | --- | --- | --- |
| a | b | c |  |
| a | "b | b" | c |

### PR DESCRIPTION
**Pull request by anazard**
Fixes #2 
It might not be perfect, but fit in my use cases, and might work for most people.
Feel free to modify or suggest any ideas you may come up with.

- Added regex to only match delimiters that are not in a sentence wrapped by quotes.

Thank you!

### CODE REVIEW

**Comment by donatj** [comment]
That regex is actually surprisingly resilient. I'm playing with it trying to come up with ways to break it so I'm aware when people come backing, but overall this looks really good! Thank you!

File: lib/CsvToMarkdown.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ function csvToMarkdown(csvContent, delimiter, hasHeader) {
         if (typeof tabularData[i] == "undefined") {
             tabularData[i] = [];
         }
-        var row = e.split(delimiter);
+        var regex = new RegExp('(?!\\B"[^"]*)' + delimiter + '(?![^"]*"\\B)');
```
Comment (donatj) [inline lib/CsvToMarkdown.js]: In trying to fully understand this regex, as well as playing with it here https://regex101.com/r/9TKbIp/1/

I'm not certain the first negative lookahead actually does anything at all?

WIthout it https://regex101.com/r/9TKbIp/2 I get the same number of matches here, and I can't seem to find a way to make the result different… 

If I'm reading the first regex correctly it's simply negatively looking ahead at the delimiter itself which is unessessary? Let me know if I am mistaken.

Comment (anazard) [inline lib/CsvToMarkdown.js]: Sorry for the delay to reply! Thanks for reviewing the PR.

Yes, you're right, having the expression both sides isn't necessary. I came up with that solution to use it on a project I was working on at the moment, but hadn't had time to play with it and test new stuff. I've tried the regex101 above with other sample strings and still got the same matches.

Comment (donatj) [inline lib/CsvToMarkdown.js]: If you would like to remove the left negative lookahead I will happily merge this.

**Comment by anazard** [comment]
Changes pushed. Thank you very much for your attention and work! 💯

**Pull request merged**